src/app/DeviceProxy.h

```diff
@@ -54,7 +55,12 @@ class DLL_EXPORT DeviceProxy
 
     virtual CHIP_ERROR SetPeerId(ByteSpan rcac, ByteSpan noc) { return CHIP_ERROR_NOT_IMPLEMENTED; }
 
-    const ReliableMessageProtocolConfig & GetRemoteMRPConfig() const { return mRemoteMRPConfig; }
+    /**
+     * Facilities for keeping track of the latest point we can expect the
+     * fail-safe to last through.  These timestamp values use the monotonic clock.
+     */
+    void SetFailSafeExpirationTimestamp(System::Clock::Timestamp timestamp) { mFailSafeExpirationTimestamp = timestamp; }
```

Comment:
This is odd to be on the DeviceProxy, but I guess there are few other spots

Reply:
Yep.  I considered just putting it on CommissioneeDeviceProxy, but the casting was slightly more annoying that way.  But I could do that.